### PR TITLE
Feature/gefs soilt fix

### PR DESCRIPTION
**Pull request by LarissaReames-NOAA**
This PR addresses Issue #447.

A unit test is created (ftst_sfc_input_data) that tests the newly-created subroutine in to which the fix for soil temperature (and other adjustments to soil temperature based on landmask) was put, and it and all other tests are passed.

### CODE REVIEW

**Comment by LarissaReames-NOAA** [comment]
I just realized that this PR includes a commit (that I did not write) to fix erroneously high canopy moisture content in GEFS data . I'll need to move that in to it's subroutine and write a quick test for that as well.

File: sorc/chgres_cube.fd/input_data.F90
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/input_data.F90:6606]: I would add more detail here. You are not just updating soil temperature, but ice temperature as well. And for open water, you are using SST as a 'filler' value, since this array is not used at those points.

File: tests/chgres_cube/ftst_sfc_input_data.F90
Comment (GeorgeGayno-NOAA) [inline tests/chgres_cube/ftst_sfc_input_data.F90:44]: In this case, 0 is open water, 1 is land, and 2 is ice.

File: sorc/chgres_cube.fd/input_data.F90
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/input_data.F90:6606]: That is the chgres_cube convention which can confuse people. The soil temperature variables only hold soil temperature at land points.

File: tests/chgres_cube/ftst_sfc_input_data.F90
Comment (LarissaReames-NOAA) [reply tests/chgres_cube/ftst_sfc_input_data.F90]: I've updated the comments in the most recent commit to address this and your previous comments.

File: sorc/chgres_cube.fd/input_data.F90
Comment (LarissaReames-NOAA) [reply sorc/chgres_cube.fd/input_data.F90]: This has been fix. I'd like to request that we start keeping a wiki or something that we can choose to get updates about when these kinds of decisions are made. I imagine, going forward, they're going to become more common and it would be helpful to know ahead of time so we can make sure any new/modified code conforms to these choices.

Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/input_data.F90]: > This has been fixed. I'd like to request that we start keeping a wiki or something that we can choose to get updates about when these kinds of decisions are made. I imagine, going forward, they're going to become more common and it would be helpful to know ahead of time so we can make sure any new/modified code conforms to these choices.

Totally agree.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Ready to merge.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Comment by LarissaReames-NOAA** [comment]
I may be showing a bit of GitHub non-expertise here, but how exactly did @edwardhartnett push to the branch on my fork without write access?

**Comment by edwardhartnett** [comment]
Magic!
image

**Comment by edwardhartnett** [comment]
@LarissaReames-NOAA there is a little button allowing us to update your branch with the latest changes to develop, which needs to happen before this PR is merged.

**Pull request merged**
